src/analysis/modal-utils.ts

```diff
@@ -97,3 +178,126 @@ export const buildExistingEnvironmentConfig = (
       : undefined,
   };
 };
+
+export interface DesiredEnvironmentParams {
+  desiredRuntimeType: string;
+  timeoutInMinutes: number;
+  deleteDiskSelected: boolean;
+  upgradeDiskSelected: boolean;
+  jupyterUserScriptUri: string | undefined;
+  isCustomSelectedImage: boolean;
+  customImageUrl: string;
+  selectedImage?: ComputeImage;
+}
+
+// extract desiredruntimeconfig into a type
+// TODO: converge type with getExistingEnvironmentConfig
+// TODO: type desired runtimetype
+export const buildDesiredEnvironmentConfig = (
```

Comment:
if these util methods are only used by GcpComputeModal, maybe we can avoid adding them to the generic modal-utils file here and add them to a more specific helper file next to the only modal using it.

Reply:
Yeah, I agree, this file needs a better name because these two methods (and the smaller helpers) are the main use case for it.